InstagramClone/View Controllers/ProfileViewController.m

```diff
@@ -0,0 +1,147 @@
+//
+//  ProfileViewController.m
+//  InstagramClone
+//
+//  Created by Angelina Zhang on 6/30/22.
+//
+
+#import "ProfileViewController.h"
+#import "ProfileView.h"
+
+#import "ParsePostHandler.h"
+#import "ParseUserHandler.h"
+#import "Post.h"
+#import "PostCollectionCell.h"
+
+#import "ErrorViewController.h"
+#import "DetailsViewController.h"
+
+@interface ProfileViewController () <ParsePostHandlerDelegate, ParseUserHandlerDelegate, ErrorViewControllerDelegate, UIImagePickerControllerDelegate, UINavigationControllerDelegate, UICollectionViewDelegate, UICollectionViewDataSource>
+
+@property (strong, nonatomic) IBOutlet ProfileView *profileView;
+@property (weak, nonatomic) IBOutlet UICollectionView *postsCollectionView;
+@property (nonatomic) NSMutableArray<Post *> *posts;
+@property (nonatomic) ParsePostHandler *postHandler;
+@property (nonatomic) ParseUserHandler *userHandler;
+
+@end
+
+@implementation ProfileViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.postsCollectionView.delegate = self;
+    self.postsCollectionView.dataSource = self;
+    
+    self.postHandler = [[ParsePostHandler alloc] init];
+    self.postHandler.delegate = self;
+    self.userHandler = [[ParseUserHandler alloc] init];
+    self.userHandler.delegate = self;
+    
+    [self fetchUserInfo];
+    [self fetchPosts];
+}
+
+- (void)fetchPosts {
+    [self.postHandler querySelfProfilePosts];
+}
+
+- (void)fetchUserInfo {
+    if (!self.username) {
```

Comment:
Hard to imagine not having a username at this point in the flow but I like checking for it since it could be missing. Here I would maybe just fail fast if it's not present since if you don't have a username something went really wrong I think. Better to not try and recover from some errors. Think about banking and health apps - if something goes wrong, shut it down.